python/tests/test_tools/test_toolset.py

```diff
@@ -304,3 +305,26 @@ def test_execute_action() -> None:
     toolset = ComposioToolSet()
     response = toolset.execute_action(Action.HACKERNEWS_GET_FRONTPAGE, {})
     assert response["successfull"]
+
+
+class EmailAddressModel(BaseModel):
+    name: str
+    email: str
+
+
+def test_execute_action_param_serialization() -> None:
```

Comment:
Consider adding test cases for:
1. Nested Pydantic models
2. Lists containing Pydantic models
3. Edge cases like circular references

This would help ensure the serialization works correctly in all scenarios.